python/hail/api2/matrixtable.py

```diff
@@ -1832,11 +1828,7 @@ def joiner(left):
                                                                 [expr._ast.to_hql()],
                                                                 '{}.{}'.format(prefix, uid), None, False)))
             elif indices == src._row_indices:
-                prefix = 'va'
-                joiner = lambda left: (
-                    MatrixTable(left._jvds.annotateVariantsTable(right._jvds.samplesKT(),
-                                                                 [expr._ast.to_hql()],
-                                                                 '{}.{}'.format(prefix, uid), None, False)))
+                return self.rows_table()[expr]
```

Comment:
sad that this is currently untested, but it's way harder without the ability to annotate keys!

Reply:
Yeah.  I think when the api settles down it would be useful to think a bit about the testing strategy.  I  think there are some easy things we can do (create some small, standard matrix and table data files with all the various key relationships (1-0, 1-1, many-1, etc.) to test all the joins, for example.)